### ISSUE TITLE
Using types moved from `SysBasic` to `Sys` breaks when both modules are used

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#19279]
In https://github.com/chapel-lang/chapel/pull/19262#issuecomment-1045082400, @mppf pointed out that my move of types like `socklen_t` from `SysBasic` to `Sys` breaks programs that use both since I relied on a `proc ... type` to implement the warning that the type had moved.  Specifically, a program like:

```chapel
use Sys;
use SysBasic;

var x: socklen_t;
```

results in:

```
$CHPL_HOME/modules/standard/Sys.chpl:258: error: symbol socklen_t is multiply defined
$CHPL_HOME/modules/standard/SysBasic.chpl:78: note: also defined as a function here (and possibly elsewhere)
```

This is arguably less of an issue for types that moved from `SysBasic` to `CTypes` since existing programs would not use both modules since `CTypes` had not existed before this point, and would hopefully move over to `CTypes` and stop using `SysBasic` (unless they relied on other elements of `SysBasic` in which case they'd need to use an `only` clause to get at them in order to not hit this same issue).

**Associated Future Test(s):**
[`test/library/packages/canCompileNoLink/testSockets.chpl`](
           https://github.com/chapel-lang/chapel/blob/main/test/library/packages/canCompileNoLink/testSockets.chpl
          ) #19345

**Comment by bradcray** [comment]
Capturing some possible ways to deal with this:

1) move these types back to SysBasic and leave it for someone else to deal with
  * doesn't seem very cool and just kicks the can down the road
  * doesn't help with the `c_float`/`c_double` cases, though I'm less worried about those as noted in the OP
2) for the time being, make these type aliases into procedures in Sys.chpl as well
   * main downside: would make the documentation not really reflect the state we want it to be in ultimately
   * but probably user code would not be affected in practice
   * we could/would change it to a type alias once we deprecate the warning overload in SysBasic.chpl.
3) adjust scopeResolve to be aware of "last resort" procs, and to ignore them preferring the non-proc symbol instead
   * a bit more involved, but could be generally useful
   * since "last resort" isn't currently user-facing, would only really affect code we were writing
4) rename these types to `c_socklen_t` now s.t. SysBasic would use the old names and warn of the change; Sys would use the new names and not conflict.
  * this would help with these cases, though `c_float`, `c_double` would still have this issue (though I'm less worried about this due to the rationale in the OP combined with thinking of SysBasic as a module whose days are numbered)

**Comment by mppf** [comment]
FWIW option 4 sounds great to me in the near term

**Comment by bradcray** [comment]
Added future showing the problem and locking in Sockets compilation in #19345 and noted it in the OP.

**Comment by mppf** [comment]
I just merged #19372 which does this

> adjust scopeResolve to be aware of "last resort" procs, and to ignore them preferring the non-proc symbol instead
